app/services/job-state.js

```diff
@@ -1,37 +1,37 @@
 import Service, { inject as service } from '@ember/service';
 import { task } from 'ember-concurrency';
+import { computed } from '@ember/object';
 import { filter, reads, gt } from '@ember/object/computed';
 
-const FINISHED_STATES = ['failed', 'canceled', 'passed'];
-const RUNNING_STATES = ['started'];
-const QUEUED_STATES = ['created', 'queued', 'booting', 'received'];
-const RUNNING_AND_FINISHED_STATES = RUNNING_STATES.concat(FINISHED_STATES);
-const DISPLAY_TIMEOUT_SEC = 30;
+export const FINISHED_STATES = ['failed', 'canceled', 'passed'];
+export const RUNNING_STATES = ['started'];
+export const QUEUED_STATES = ['created', 'queued', 'booting', 'received'];
+export const UNFINISHED_STATES = QUEUED_STATES.concat(RUNNING_STATES);
 
 export default Service.extend({
   store: service(),
 
-  jobs: reads('fetchJobs.lastSuccessful.value'),
-  jobsLoaded: gt('fetchJobs.performCount', 0),
+  jobs: reads('peekJobs.lastSuccessful.value'),
+  jobsLoaded: gt('fetchUnfinishedJobs.performCount', 0),
+  sortedJobs: computed('jobs', function () {
+    const { jobs } = this;
+    return jobs && jobs.sortBy('number');
+  }),
+
+  runningJobs: filter('sortedJobs.@each.state', (job) => RUNNING_STATES.includes(job.state)),
+  queuedJobs: filter('sortedJobs.@each.state', (job) => QUEUED_STATES.includes(job.state)),
+  unfinishedJobs: computed('queuedJobs.[]', 'runningJobs.[]', function () {
+    const { queuedJobs = [], runningJobs = [] } = this;
```

Comment:
This may be a bit fragile, because if a property value is `null` and not `undefined` (which is an often case for computed props), it won't be assigned default value. The `value || defaultValue` approach works more reliable in this sense 😄

Reply:
Good point! Change made